core/templates/dev/head/pages/exploration_player/ConversationSkinDirective.js

```diff
@@ -755,6 +761,13 @@ oppia.directive('conversationSkin', [
           };
 
           $scope.showUpcomingCard = function() {
+            /* This is because when answerIsCorrect becomes false below, if
+               pendingCardWasSeenBefore is true (going forward to the card from
```

Comment:
I don't think I understand the "(going forward to the card from which redirection just happened)" bit. Could you please elaborate on that? If you have card A which leads to a pending card B, then I'm interpreting that phrase as going forward from A to A, which confuses me.

Thanks!

Reply:
I meant if A->B->C was arrangement and C redirected to A, then after that, B and C are visited cards for him.

Reply:
Ah ok. Maybe add this info to the comment? The new explanation is clearer.

Reply:
Ok, will add that